types/reserved-words/index.d.ts

```diff
@@ -0,0 +1,36 @@
+// Type definitions for reserved-words 0.1
+// Project: https://github.com/zxqfox/reserved-words#readme
+// Definitions by: Spencer Miskoviak <https://github.com/skovy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+declare namespace reserved {
```

Comment:
I don't think we need a namespace here since `module.exports` isn't written directly it's assigned but the only member is `check`.

eg. you can `export` everything directly.

```
export type Dialect = ...`;

expect function check...
```

Also you missed the `KEYWORDS` export.